ui-admin/src/portal/siteContent/SiteContentLoader.tsx

```diff
@@ -0,0 +1,84 @@
+import { PortalEnvContext } from '../PortalRouter'
+import React, { useEffect, useState } from 'react'
+import { ApiContextT, SiteContent } from '@juniper/ui-core'
+import Api from 'api/api'
+import { Store } from 'react-notifications-component'
+import { failureNotification, successNotification } from 'util/notifications'
+import LoadingSpinner from 'util/LoadingSpinner'
+import SiteContentEditor from './SiteContentEditor'
+
+
+/** logic for loading, changing, and saving SiteContent objects */
+const SiteContentLoader = ({ portalEnvContext }: {portalEnvContext: PortalEnvContext}) => {
+  const { portal, portalEnv } = portalEnvContext
+  const portalShortcode = portal.shortcode
+  const [isLoading, setIsLoading] = useState(true)
+  const [siteContent, setSiteContent] = useState(portalEnv.siteContent)
+  if (!siteContent) {
+    return <div>no site content configured</div>
+  }
+
+  /** uses the admin image retrieval endpoint */
+  const getImageUrl = (cleanFileName: string, version: number) =>
+      `/api/public/portals/v1/${portalShortcode}/env/${portalEnv.environmentName}` +
+      `/siteImages/${version}/${cleanFileName}`
+  const previewApi: ApiContextT = {
+    getImageUrl,
+    submitMailingListContact: () => Promise.resolve({})
+  }
+
+  const loadSiteContent = async (stableId: string, version: number) => {
+    setIsLoading(true)
+    Api.getSiteContent(portalShortcode, stableId, version).then(response => {
+      setSiteContent(response)
+      setIsLoading(false)
+    }).catch(() => {
+      Store.addNotification(failureNotification('Could not load site content'))
+    })
+  }
+
+  /** saves the current content as a new version, and updates the portal environment to use it */
+  const createNewVersion = async (workingContent: SiteContent) => {
+    let newVersion: SiteContent
+    try {
+      newVersion = await Api.createNewSiteContentVersion(portalShortcode, workingContent.stableId, workingContent)
+      Store.addNotification(successNotification(`Version ${newVersion.version} created`))
+    } catch {
+      Store.addNotification(failureNotification('save failed'))
+      return
+    }
+    try {
+      let updatedEnv = {
+        ...portalEnv,
+        siteContentId: newVersion.id
+      }
+      updatedEnv = await Api.updatePortalEnv(portalShortcode, portalEnv.environmentName, updatedEnv)
```

Comment:
Suggestion: We could have an API endpoint that both creates a new version and updates the portal env. We would have more control over the transaction and be resilient to client connection issues. As is, if the first call succeeds and the second fails, unless there's a UI allowance for just the `updatePortalEnv` part, then we'd probably have to intervene to fix it.